3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/PiiPseudonymizerEnrichment.scala

```diff
@@ -0,0 +1,237 @@
+          if (outputField == fieldName) =>
+        (inputField, ((arg1: String, arg2: String) => tf.tupled.andThen(_.map(applyStrategy))((arg1, arg2)), outputField))
+    }
+}
+
+/**
+ * PiiStrategy trait. This corresponds to a strategy to apply to a single field. Currently only only String input is
+ * supported.
+ */
+sealed trait PiiStrategy {
+  def scramble: String => String
+}
+
+/**
+ * Companion object. Lets us create a PiiPseudonymizerEnrichment
+ * from a JValue.
+ */
+object PiiPseudonymizerEnrichment extends ParseableEnrichment {
+  val supportedSchema = SchemaCriterion("com.snowplowanalytics.snowplow.enrichments", "pii_enrichment_config", "jsonscehma", 1, 0, 0)
+
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[PiiPseudonymizerEnrichment] = {
+    for {
+      conf <- matchesSchema(config, schemaKey)
+      enabled = ScalazJson4sUtils.extract[Boolean](conf, "enabled").toOption.getOrElse(false)
+      piiFields        <- ScalazJson4sUtils.extract[List[JObject]](conf, "parameters", "pii").leftMap(_.getMessage)
+      strategyFunction <- extractStrategyFunction(config)
+      hashFunction     <- getHashFunction(strategyFunction)
+      piiFieldList     <- extractFields(piiFields, PiiStrategyPseudonymize(hashFunction))
+    } yield if (enabled) PiiPseudonymizerEnrichment(piiFieldList) else PiiPseudonymizerEnrichment(List())
+  }.leftMap(_.toProcessingMessageNel)
+
+  private def getHashFunction(strategyFunction: String) =
+    try {
+      MessageDigest.getInstance(strategyFunction).success
+    } catch {
+      case e: NoSuchAlgorithmException =>
+        s"Could not parse PII enrichment config: ${e.getMessage()}".fail
+    }
+
+  private def extractFields(piiFields: List[JObject], strategy: PiiStrategy) =
+    piiFields.map {
+      case JObject(List(("pojo", JObject(List(("field", JString(fieldName))))))) => PiiPojo(strategy, fieldName).success
+      case JObject(List(("json", jsonField))) =>
+        val fieldsDisj: \/[MappingException, List[String]] = List(extractString(jsonField, "field"),
+                                                                  extractString(jsonField, "schemaCriterion"),
+                                                                  extractString(jsonField, "jsonPath")).sequenceU
+        fieldsDisj.validation
+          .leftMap(_ => "Configuration file has unexpected structure")
+          .flatMap[String, PiiJson] { (l: List[String]) =>
+            SchemaCriterion
+              .parse(l(1))
+              .rightMap { sc =>
+                PiiJson(strategy, l(0), sc, l(2))
+              }
+              .leftMap(_.getMessage)
+          }
+      case x => "Configuration file has unexpected structure".failure
+    }.sequenceU
+
+  implicit val json4sFormats = DefaultFormats
+
+  private def extractString(jValue: JValue, field: String): \/[MappingException, String] =
+    \/.fromTryCatchThrowable[String, MappingException] {
```

Comment:
shouldn't this be `fromTryCatchNonFatal`?

Reply:
I could not find that in this version of \/ but it's actually more specific than than NonFatal.